sympy/physics/control/lti.py

```diff
@@ -249,6 +252,112 @@ def expand(self):
+        obtained by taking the limit as ``var`` goes to zero.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a, b
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> tf1 = TransferFunction(s + 3, s**2 - 9, s)
+        >>> tf1.dcgain()
+        -1/3
+        >>> tf2 = TransferFunction(p**2, p - 3 + p**3, p)
+        >>> tf2.dcgain()
+        0
+        >>> tf3 = TransferFunction(a*p**2 - b, s + b, s)
+        >>> tf3.dcgain()
+        (a*p**2 - b)/b
+
+        """
+        m = Mul(self.num, Pow(self.den, -1, evaluate=False), evaluate=False)
+        return limit(m, self.var, 0)
+
+    def pole(self):
+        """
+        Returns the poles of a transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> tf1 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+        >>> tf1.pole()
+        [1, -5]
+        >>> tf2 = TransferFunction((1 - s)**2, (s**2 + 1)**2, s)
+        >>> tf2.pole()
+        [-I, -I, I, I]
+        >>> tf3 = TransferFunction(s**2, a*s + p, s)
+        >>> tf3.pole()
+        [-p/a]
+
+        """
+        if self.den.is_number:
+            return []
+        p = Poly(self.den, self.var)
+        poles = None
+        try:
+            cmp_matrix = Matrix.companion(p)
+        except ValueError:
+            poles = roots(p, self.var)
+
+        if not poles:
+            try:
+                poles = cmp_matrix.eigenvals()
+            except MatrixError:
+                raise ValueError("Could not compute poles of this transfer function.")
+
+        rts = []
+        for key, val in poles.items():
+            rts.append(key) if val == 1 else rts.extend([key]*val)
+
+        return rts
+
+    def zero(self):
+        """
+        Returns the zeros of a transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> tf1 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+        >>> tf1.zero()
+        [1, -3]
+        >>> tf2 = TransferFunction((1 - s)**2, (s**2 + 1)**2, s)
+        >>> tf2.zero()
+        [1, 1]
+        >>> tf3 = TransferFunction(s**2, a*s + p, s)
+        >>> tf3.zero()
+        [0, 0]
+
+        """
+        if self.num.is_number:
+            return []
+        p = Poly(self.num, self.var)
+        zeros = None
+        try:
+            cmp_matrix = Matrix.companion(p)
+        except ValueError:
+            zeros = roots(p, self.var)
+
+        if not zeros:
+            try:
+                zeros = cmp_matrix.eigenvals()
+            except MatrixError:
+                raise ValueError("Could not compute zeros of this transfer function.")
+
+        rts = []
+        for key, val in zeros.items():
+            rts.append(key) if val == 1 else rts.extend([key]*val)
```

Comment:
This looks like an unencessary micro-optimziation
```suggestion
            rts.extend([key]*val)
```

Reply:
Thanks 😅